sql/driver/driver_test.go

```diff
@@ -62,7 +61,11 @@ func readAll(rows *sql.Rows) ([][]string, error) {
 		}
 		rowStrings := make([]string, 0, len(cols))
 		for _, v := range vals {
-			rowStrings = append(rowStrings, fmt.Sprint(v))
+			if val := *v.(*interface{}); val != nil {
```

Comment:
That cast is real ugly. I think it might be more obvious to create a separate `valptrs` slice. Something like:

```
  vals := make([]interface{}, len(cols))
  valptrs := make([]interface{}, len(vals))
  for i := range vals {
    valptrs[i] = &vals[i]
  }

  for rows.Next() {
    if err := rows.Scan(valptrs...); err != nil {
      ...
    }
    for _, v := range vals {
      if v != nil {
        rowStrings = append(rowStrings, fmt.Sprint(v))
      } else {
        rowStrings = append(rowStrings, "NULL")
      }
    }
  }
```